tesseract-server/src/handlers/logic_layer/geoadjacents.rs

```diff
@@ -0,0 +1,351 @@
+    let geoservice_url = &req.state().env_vars.geoservice_url;
+
+    let dimensions_map = match getdimensions(&cuts_map, &cube, &cube_cache, &level_map, &property_map, &geoservice_url) {
+        Ok(dm) => dm,
+        Err(err) => return Ok(HttpResponse::NotFound().json(err.to_string()))
+    };
+
+    if dimensions_map.len() == 0{
+        return Ok(HttpResponse::NotFound().json("Unable to genrate result".to_string()));
+    }
+
+    let final_headers: Vec<String> = ["level".to_string(), "id".to_string(), "operation".to_string(), "value".to_string()].to_vec();
+    let mut final_columns: Vec<Column> = vec![];
+
+    let mut col_0: Vec<String> = Vec::new();
+    let mut col_1: Vec<String> = Vec::new();
+    let mut col_2: Vec<String> = Vec::new();
+    let mut col_3: Vec<String> = Vec::new();
+
+    for row in dimensions_map{
+        col_0.push(row.get(0).unwrap().to_string());
+        col_1.push(row.get(1).unwrap().to_string());
+        col_2.push(row.get(2).unwrap().to_string());
+        col_3.push(row.get(3).unwrap().to_string());
+    }
+
+    final_columns.push(Column{
+        name: "placeholder".to_string(),
+        column_data: ColumnData::Text(col_0)
+    });
+    final_columns.push(Column{
+        name: "placeholder".to_string(),
+        column_data: ColumnData::Text(col_1)
+    });
+    final_columns.push(Column{
+        name: "placeholder".to_string(),
+        column_data: ColumnData::Text(col_2)
+    });
+    final_columns.push(Column{
+        name: "placeholder".to_string(),
+        column_data: ColumnData::Text(col_3)
+    });
+
+    let final_df = DataFrame { columns: final_columns };
+
+    let content_type = util::format_to_content_type(&format);
+
+    match format_records(&final_headers, final_df, format) {
+        Ok(res) => {
+            Ok(HttpResponse::Ok()
+                .set(content_type)
+                .body(res))
+        },
+        Err(err) => Ok(HttpResponse::NotFound().json(err.to_string())),
+    }
+}
+
+
+pub fn getdimensions(
+    cuts_map: &HashMap<String, String>,
+    cube: &Cube,
+    cube_cache: &CubeCache,
+    level_map: &HashMap<String, LevelName>,
+    property_map: &HashMap<String, Property>,
+    geoservice_url: &Option<Url>
+) -> Result<Vec<Vec<String>>, Error> {
+
+    let mut dimensions_map: Vec<Vec<String>> = vec![];
+
+    let mut level_matches: Vec<LevelName> = vec![];
+
+    for (cut_key, cut_values) in cuts_map.iter(){
+        if cut_values.is_empty(){
+            continue;
+        }
+        let element:Vec<String> = cut_values.split(":").map(|s| s.to_string()).collect();
+        let cut_key_values:Vec<String> = match element.get(0){
+            Some(ckv) => ckv.split(",").map(|s| s.to_string()).collect(),
+            None => continue,
+        };
+        let operations:Vec<String> = match element.get(1){
+            Some(op) => op.split(",").map(|s| s.to_string()).collect(),
+            None => continue,
+        };
```

Comment:
For consistency and ease of reading, we should add spaces before variable types. Similarly, we've been using spaces before open curly braces:

`let operations:Vec<String> = match element.get(1){` becomes `let operations: Vec<String> = match element.get(1) {`

We should set up a linter for this type of stuff, but we've been doing it manually for now...